workflow/workflow.go

```diff
@@ -325,31 +349,91 @@ func (l *loopState) onStageComplete(stepID string, previousStage *string, previo
 	l.notifySteps()
 }
 
+// Marks the outputs of that stage unresolvable.
+// Optionally skip an output that should not unresolve.
+// Does not (un)resolve the non-output node for that stage.
+// To prevent a deadlock, this `notifySteps()` should be called at some point after this is called.
```

Comment:
to make the implicit explicit, do you mean the caller of this function is responsible for a subsequent call `notifySteps()`?

Reply:
The caller needs to call `notifySteps()` at some point after it.